docs/cluster-configuration.md

```diff
@@ -150,3 +150,70 @@ spec:
     withHostModelCPU: true
     withHostPassthroughCPU: true
 ```
+
+## Configurations via Annotations
+In addition to `featureGates` field in HCO CR's spec, the user can set annotations in the HCO CR to unfold more configuration options.
```

Comment:
You should warn the reader that annotations are less formal and may be dropped without the same deprecation process of normal API

Reply:
added warning.